include/veriblock/pop/blockchain/pop/fork_resolution.hpp

```diff
@@ -444,33 +444,35 @@ struct PopAwareForkResolutionComparator {
                                        protecting_params_t,
                                        protected_params_t>;
 
-  PopAwareForkResolutionComparator(std::shared_ptr<ProtectingBlockTree> tree,
+  PopAwareForkResolutionComparator(ProtectedBlockTree& ed,
```

Comment:
What was the intention to move ProtectedBlockTree& ed as an argument of the functions to the field of this struct ? What benefits these should bring to us ?

Reply:
This is just a clean up, so the benefits are small. Basically, the states of the protected and protecting trees are coupled. Any pop state machine call on the protected tree changes the state of the protecting tree. So there's no way we could have 2 protected trees per 1 protecting tree. This change leads to a minor simplification of code.